eta/core/datasets.py

```diff
@@ -1372,7 +1381,27 @@ def transform(self, src):
             raise DatasetTransformerError()
         old_records = src.records
         src.clear()
-        # @TODO impl me! - Also might want to throw error if data is not video.
+        for record in old_records:
+            start_frame = record.clip_start_frame
+            while start_frame <= record.clip_end_frame:
+                end_frame = start_frame + self.clip_len - 1
+                if end_frame > record.clip_end_frame:
+                    end_frame = record.clip_end_frame
+                    clip_duration = int(end_frame - start_frame + 1)
+                    if clip_duration < int(self.min_clip_len):
+                        break
+                self.add_clipping(start_frame, end_frame, record, src.records)
+                start_frame += self.clip_len
+                start_frame += self.stride
+
+    def add_clipping(self, start_frame, end_frame, old_record, records):
+        frame_rate = old_record.frame_rate
+        new_record = old_record.copy()
```

Comment:
Thank you!

Reply:
> > Should we not allow users to clip to 1 frame.
> > I can prevent them in the front end.
> 
> I would say it's not necessary. If that is what they set as their minimum, then so be it.

but it seems like videos of 1 frame don't play

Reply:
Okay, if you think it will avoid problems like that, then sure. I don't have a strong opinion either way.